app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/teidata/teievents/StageViewHolder.kt

```diff
@@ -93,7 +95,7 @@ internal class StageViewHolder(
                             },
                     ) {
                         NewEventOptions(presenter.getNewEventOptionsByStages(stage)) {
-                            presenter.onAddNewEventOptionSelected(it, stage)
+                            singleEventEnforcer.processEvent { presenter.onAddNewEventOptionSelected(it, stage) }
```

Comment:
Same here, could this be handled in the presenter?

Reply:
I agree and have updated this